Codeinwp/neve
# fix: infinite scroll on blog with masonry enabled [closes #2373]

### Summary
- fixes issue with infinite scroll & masonry where post fetching and adding to DOM;
- fixes issue with infinite scroll where if after fetching posts, the loader was still on screen, no other posts were loading;


### Will affect visual aspect of the product

NO

### Test instructions


- Set the blog layout to 3 columns, make sure you have 8 posts in the same category;
- Enable infinite scroll + masonry on the blog;
- Set the reading settings so that the blog page displays only 6 posts/page;
- Go to the respective category page - the posts should be properly loading (all of them should be on page after the content is fetched)
- Everything should work as expected on the blog 

--- 

- Set the blog layout to 2 columns
- Enable infinite scroll ( should work fine both with & without masonry)
- Set the reading settings so that the blog page displays only 2 posts/page;
- Go to the blog page, after loading the first page, if the loader is still on screen, the next page should be loaded -> you can achieve this by zooming out so that the page can display multiple blog posts on the first load. It should be loading enough to fill the page and then stop until the loader is visible again in the viewport;


Closes #2373 and the freshly found issue with posts not loading if the loader is still on screen.
